File: ventas/VentasDepartamentos.java
import java.util.*;

public class VentasDepartamentos {
    static String[] meses = {"Enero", "Febrero", "Marzo", "Abril", "Mayo", "Junio",
                             "Julio", "Agosto", "Septiembre", "Octubre", "Noviembre", "Diciembre"};
    static ArrayList<String> departamentos = new ArrayList<>();
    static ArrayList<int[]> ventas = new ArrayList<>();
    static Scanner sc = new Scanner(System.in);

    // Inicializar con 3 departamentos base
    static {
        departamentos.add("Ropa");
        departamentos.add("Deportes");
        departamentos.add("Juguetería");
        for (int i = 0; i < 12; i++) {
            ventas.add(new int[departamentos.size()]);
        }
    }

    // Agregar nuevo departamento
    public static void agregarDepartamento() {
        System.out.print("Ingresa el nombre del nuevo departamento: ");
        String nuevo = sc.next();
        departamentos.add(nuevo);

        // Expandir las filas (meses) para incluir el nuevo departamento
        for (int i = 0; i < 12; i++) {
            int[] filaVieja = ventas.get(i);
            int[] filaNueva = Arrays.copyOf(filaVieja, departamentos.size());
            ventas.set(i, filaNueva);
        }
        System.out.println(" Departamento agregado: " + nuevo);
    }

    // Insertar o actualizar venta
    public static void insertarVenta() {
        int mes = pedirMes();
        int dep = pedirDepartamento();
        System.out.print("Ingresa la venta: ");
        ventas.get(mes)[dep] = sc.nextInt();
        System.out.println(" Venta registrada.");
    }

    // Buscar venta
    public static void buscarVenta() {
        int mes = pedirMes();
        int dep = pedirDepartamento();
        System.out.println("La venta es: " + ventas.get(mes)[dep]);
    }

    // Eliminar venta
    public static void eliminarVenta() {
        int mes = pedirMes();
        int dep = pedirDepartamento();
        ventas.get(mes)[dep] = 0;
        System.out.println(" Venta eliminada.");
    }

    // Mostrar tabla completa
    public static void mostrarVentas() {
        System.out.println("\n--- Tabla de Ventas ---");
        System.out.print("Mes\t\t");
        for (String d : departamentos) {
            System.out.print(d + "\t");
        }
        System.out.println();
        for (int i = 0; i < 12; i++) {
            System.out.print(meses[i] + "\t");
            int[] fila = ventas.get(i);
            for (int v : fila) {
                System.out.print(v + "\t\t");
            }
            System.out.println();
        }
    }

    private static int pedirMes() {
        System.out.print("Ingresa el número de mes (1-12): ");
        return sc.nextInt() - 1;
    }

    private static int pedirDepartamento() {
        System.out.println("Departamentos:");
        for (int i = 0; i < departamentos.size(); i++) {
            System.out.println((i + 1) + ". " + departamentos.get(i));
        }
        System.out.print("Elige un departamento: ");
        return sc.nextInt() - 1;
    }

    public static void main(String[] args) {
        int opcion;
        do {
            System.out.println("\n--- Menú ---");
            System.out.println("1. Insertar/Actualizar venta");
            System.out.println("2. Buscar venta");
            System.out.println("3. Eliminar venta");
            System.out.println("4. Mostrar tabla de ventas");
            System.out.println("5. Agregar nuevo departamento");
            System.out.println("6. Salir");
            System.out.print("Elige una opción: ");
            opcion = sc.nextInt();

            switch (opcion) {
                case 1 -> insertarVenta();
                case 2 -> buscarVenta();
                case 3 -> eliminarVenta();
                case 4 -> mostrarVentas();
                case 5 -> agregarDepartamento();
                case 6 -> System.out.println("Saliendo...");
                default -> System.out.println("Opción inválida.");
            }
        } while (opcion != 6);
    }
}

File: ventas/VentasDepartamentos.py
class VentasDepartamentos:
    def __init__(self):
        self.meses = ["Enero", "Febrero", "Marzo", "Abril", "Mayo", "Junio",
                      "Julio", "Agosto", "Septiembre", "Octubre", "Noviembre", "Diciembre"]
        self.departamentos = ["Ropa", "Deportes", "Juguetería"]
        self.ventas = [[0 for _ in range(len(self.departamentos))] for _ in range(12)]

    def insertar_venta(self, mes, dep, valor):
        self.ventas[mes][dep] = valor

    def buscar_venta(self, mes, dep):
        return self.ventas[mes][dep]

    def eliminar_venta(self, mes, dep):
        self.ventas[mes][dep] = 0

    def mostrar_ventas(self):
        print("\n--- Tabla de Ventas ---")
        print("Mes\t\t" + "\t".join(self.departamentos))
        for i in range(12):
            fila = [str(self.ventas[i][j]) for j in range(len(self.departamentos))]
            print(f"{self.meses[i]}\t" + "\t\t".join(fila))

    def agregar_departamento(self, nombre):
        nombre = nombre.strip()
        if not nombre:
            print(" Nombre inválido.")
            return
        if nombre in self.departamentos:
            print(" Ya existe el departamento:", nombre)
            return
        self.departamentos.append(nombre)
        for fila in self.ventas:
            fila.append(0)
        print(f" Departamento agregado: {nombre}")


def pedir_entero(rango_min, rango_max, prompt=""):
    while True:
        try:
            valor = int(input(prompt))
            if rango_min <= valor <= rango_max:
                return valor
            print(f"Ingresa un número entre {rango_min} y {rango_max}.")
        except ValueError:
            print("Ingresa un número válido.")

def elegir_mes(app):
    return pedir_entero(1, 12, "Ingresa el número de mes (1-12): ") - 1

def elegir_dep(app):
    print("\nDepartamentos:")
    for i, d in enumerate(app.departamentos, start=1):
        print(f"{i}. {d}")
    return pedir_entero(1, len(app.departamentos), "Elige un departamento: ") - 1

ventas = VentasDepartamentos()

while True:
    print("\n--- Menú ---")
    print("1. Insertar venta")
    print("2. Buscar venta")
    print("3. Eliminar venta")
    print("4. Mostrar tabla de ventas")
    print("5. Agregar nuevo departamento")
    print("6. Salir")

    opcion = pedir_entero(1, 6, "Elige una opción: ")

    if opcion == 1:
        mes = elegir_mes(ventas)
        dep = elegir_dep(ventas)
        valor = pedir_entero(0, 10**9, "Ingresa la venta: ")
        ventas.insertar_venta(mes, dep, valor)
        print(" Venta registrada.")

    elif opcion == 2:
        mes = elegir_mes(ventas)
        dep = elegir_dep(ventas)
        print(f"La venta es: {ventas.buscar_venta(mes, dep)}")

    elif opcion == 3:
        mes = elegir_mes(ventas)
        dep = elegir_dep(ventas)
        ventas.eliminar_venta(mes, dep)
        print(" Venta eliminada.")

    elif opcion == 4:
        ventas.mostrar_ventas()

    elif opcion == 5:
        nombre = input("Nombre del nuevo departamento: ")
        ventas.agregar_departamento(nombre)

    elif opcion == 6:
        print("Saliendo...")
        break

